### ISSUE TITLE
Six (any version) doesn't work on Sublime Text 3189 due to recent ST API changes

### ISSUE THREAD

**Issue by slice** [guillermooo/Six#240]
Version: v0.8.106-dev.20190122.1

Simple motions (hjkl) don't work at all. This error is constantly printed in the console when trying to do anything related to Six:

```
2019-02-01 21:16:40,665 - CRITICAL - Six.lib.editor_internal - Unexpected error.
2019-02-01 21:16:40,665 - ERROR - Six.lib.editor_internal - 'NoneType' object has no attribute 'has_non_empty_selection'
  File "D:/a/1/six-release\lib\editor_internal.py", line 92, in handle
  File "D:/a/1/six-release\lib\editor_internal.py", line 182, in _handle_state
  File "D:/a/1/six-release\lib\command_state.py", line 852, in _handle
  File "D:/a/1/six-release\lib\command_state.py", line 432, in process
2019-02-01 21:16:40,665 - INFO - Six.lib.editor_internal - Trying to reenter Normal mode...
2019-02-01 21:16:40,665 - CRITICAL - Six.lib.editor_internal - Could not reenter Normal mode. Please restart Sublime Text.
  File "D:/a/1/six-release\lib\editor_internal.py", line 224, in _handle_exception
  File "D:/a/1/six-release\lib\command_state.py", line 951, in enter_mode
  File "D:/a/1/six-release\lib\command_state.py", line 425, in enter
  File "D:/a/1/six-release\lib\command_state_internal.py", line 128, in enter
  File "D:/a/1/six-release\lib\command_state_internal.py", line 38, in enter
  File "D:/a/1/six-release\lib\command_state_internal.py", line 50, in _esc
2019-02-01 21:16:40,883 - CRITICAL - Six.lib.editor_internal - Unexpected error.
2019-02-01 21:16:40,883 - ERROR - Six.lib.editor_internal - 'NoneType' object has no attribute 'has_non_empty_selection'
  File "D:/a/1/six-release\lib\editor_internal.py", line 92, in handle
  File "D:/a/1/six-release\lib\editor_internal.py", line 182, in _handle_state
  File "D:/a/1/six-release\lib\command_state.py", line 852, in _handle
  File "D:/a/1/six-release\lib\command_state.py", line 432, in process
2019-02-01 21:16:40,884 - INFO - Six.lib.editor_internal - Trying to reenter Normal mode...
2019-02-01 21:16:40,884 - CRITICAL - Six.lib.editor_internal - Could not reenter Normal mode. Please restart Sublime Text.
  File "D:/a/1/six-release\lib\editor_internal.py", line 224, in _handle_exception
  File "D:/a/1/six-release\lib\command_state.py", line 951, in enter_mode
  File "D:/a/1/six-release\lib\command_state.py", line 425, in enter
  File "D:/a/1/six-release\lib\command_state_internal.py", line 128, in enter
  File "D:/a/1/six-release\lib\command_state_internal.py", line 38, in enter
  File "D:/a/1/six-release\lib\command_state_internal.py", line 50, in _esc
```

**Comment by guillermooo** [comment]
I also get this after updating to the latest version of Sublime Text, but I haven't seen it before.

**Comment by guillermooo** [comment]
As far as I can tell, the `plugin_loaded()` function that starts Six isn't called from the `.pyc` file, but it works fine from the `.py` file. This seems to be happening in the newest dev build. The release notes state:

>BUILD 3189
1 February 2019
Mini diffs now indicate inserted and deleted lines
Added Goto/Next Modification and Goto/Previous Modification commands
The mini_diff setting can now be set to "auto", which will only show mini diffs for files in Git repositories
Added the git_diff_target setting, to control if the mini diff for Git files is against the index or HEAD
Git repositories are now detected through symlinks
Fixed a bug calculating Git status badges in the side bar
Fixed a caret positioning bug when non-trivial graphemes are present
Windows: Fixed a bug where auto complete entries would contain an ellipsis when not required
Windows: Fixed a rendering issue with certain combining characters
Windows: Fixed some fonts having an incorrect baseline
Windows: Properly unlock directories after contained files are closed
**API: Various minor improvements related to plugin module loading and unloading**

So, currently I'm trying to figure out if this is the source of the issue.

**Comment by guillermooo** [comment]
@slice Confirmed Six is working in dev build 3188. So the new API changes in 3189 have broken Six (release/dev builds). Six still works fine locally for me from the .py files (any ST version).

**Comment by guillermooo** [comment]
If you could confirm also that you are using ST 3189, I will change the issue's title to better reflect the problem.

**Comment by slice** [comment]
Yep, I am.

**Comment by guillermooo** [comment]
Reported to Sublime HQ as https://github.com/SublimeTextIssues/Core/issues/2609

**Comment by guillermooo** [comment]
Should be fixed by build [0.8.106-dev.20190202.1](https://github.com/guillermooo/Six/blob/master/messages/0.8.106-dev.20190202.1.txt).

**Comment by slice** [comment]
Thanks for the quick fix.
